docs/UsingCppInterOp.rst

```diff
@@ -65,35 +65,255 @@ robust (simple function calls, no inheritance, etc.). The goal is to make it as
+interactive languages.
+
+The goal of these enhancements is to eventually land these interoperability
+tools (including CppInterOp) to greater communities like LLVM and Clang, to
+enable C++ to interact with other languages besides Python.
+
+Example: Template Instantiation
+-------------------------------
+
+The developmental cppyy version can run basic examples such as the one
+here. Features such as standalone functions and basic classes are also
+supported.
+
+C++ code (Tmpl.h)
+
+::
+
+   template <typename T>
+   struct Tmpl {
+     T m_num;
+     T add (T n) {
+       return m_num + n;
+   }
+   };
+
+Python Interpreter
+
+::
+
+   >>> import cppyy
+   >>> import cppyy.gbl as Cpp
+   >>> cppyy.include("Tmpl.h")
+   >>> tmpl = Tmpl[int]()
+   >>> tmpl.m_num = 4
+   >>> print(tmpl.add(5))
+   9
+   >>> tmpl = Tmpl[float]()
+   >>> tmpl.m_num = 3.0
+   >>> print(tmpl.add(4.0))
+   7.0
+
+Where does the cppyy code reside?
+---------------------------------
+
+Following are the main components where cppyy logic (with Compiler Research
+Organization’s customizations started by `sudo-panda`_) resides:
+
+-  `cppyy <https://github.com/compiler-research/cppyy>`_
+-  `cppyy-backend <https://github.com/compiler-research/cppyy-backend>`_
+-  `CPyCppyy <https://github.com/compiler-research/CPyCppyy>`_
+
+..
+
+   Note: These are forks of the `upstream cppyy`_ repos created by `wlav`_.
+
+CppInterOp is a separate library that helps these packages communicate with C++
+code.
+
+-  `CppInterOp <https://github.com/compiler-research/CppInterOp/tree/main>`_
 
-**LLVM Integration:** The CppInterOp interfaces will be part of the LLVM 
-toolchain (as part of Clang-REPL).
+How cppyy components interact with each other
+---------------------------------------------
 
-**Better C++ Support:** C++ features such as Partial Template Specialisation will
-be available through CppInterOp.
+cppyy is made up of the following packages: 
 
-**Fewer Lines of Code:** A lot of dependencies and workarounds will be removed,
-reducing the lines of code required to execute CPPYY.
+- A frontend: cppyy, 
 
-**Well tested interoperability Layer:** The CppInterOp interfaces have full
-unit test coverage.
+- A backend: cppyy-backend, and 
 
+- An extension: CPyCppyy.
+
+Besides these, the ``CppInterOp`` library serves as an additional layer on top
+of Cling/Clang-REPL that helps these packages in communicating with C++ code.
+
+**1. cppyy-backend**
+
+The `cppyy-backend`_ pacakge forms a layer over ``cppyy``, for example,
+modifying some functionality to provide the functions required for
+``CPyCppyy``. 
+
+  `CPyCppyy`_ is a CPython extension module built on top of the same backend
+  API as PyPy/_cppyy. It thus requires the installation of the cppyy-backend
+  for use, which will pull in Cling. 
+
+``cppyy-backend`` also adds some `utilities`_ to help with repackaging and
+redistribution.
+
+For example, it initializes the interpreter (using the
```

Comment:
given the paragraph break the meaning of "it" is totally unclear here.

Reply:
updated:
For example, ``cppyy-backend`` it initializes the interpreter...